src/libmeasurement_kit/report/report.cpp

```diff
@@ -44,7 +44,41 @@ void Report::open(Callback<Error> callback) {
     }), callback);
 }
 
-void Report::write_entry(Entry entry, Callback<Error> callback) {
+void Report::write_entry(Entry entry, Callback<Error> callback,
+                         Var<Logger> logger) {
+    if (report_id == "") {
+        auto count = 0;
+        for (auto &reporter : reporters_) {
+            auto rid = reporter->get_report_id();
+            if (rid != "") {
+                report_id = rid;
+                ++count;
+            }
+        }
+        /*
+         * XXX For now we trust the last reporter that provides us with a
+         * non-zero-length report-id. The reason for doing the full scan
+         * over the existing reporters and for asserting is to be ready for
+         * an hypothetical moment where more than one reporter will give
+         * back a good report-id (only ooni_reporter does that).
+         */
+        assert(count <= 1);
```

Comment:
I think it's excessive to use an assert here. I would prefer to return an error in this case, because this is for me not a case the library (and so the APP) should crash. What do you think about? Maybe we can add some unit test to be sure that in the future this case will never happens.